New code example for the "var pattern" (issue #12736) (#17197)
dotnet/docs: docs/csharp/language-reference/keywords/is.md

### var pattern

A pattern match with the `var` pattern always succeeds. Its syntax is:

```csharp
   expr is var varname
```

> where the value of *expr* is always assigned to a local variable named *varname*. *varname* is a variable of the same type as the compile time type of *expr*. 

Reviewer: ```suggestion
where the value of *expr* is always assigned to a local variable named *varname*. *varname* is a variable of the same type as the compile-time type of *expr*. 
```

> If *expr* is `null`, the `is` expression is still true and assigns `null` to *varname*. 

Reviewer: ```suggestion
If *expr* evaluates to `null`, the `is` expression produces `true` and assigns `null` to *varname*. 
```

> If *expr* evaluates to `null`, the `is` expression produces `true` and assigns `null` to *varname*. 

Reviewer 2: Is it worth adding a clarifying comment here?  In most situations `is` produces `false` for `null`:

```suggestion
If *expr* evaluates to `null`, the `is` expression produces `true` and assigns `null` to *varname*.  The var pattern is one of the few uses of `is` that produces `true` for a `null` value.
```

Reviewer 3: I don't think it's particularly useful. Side question, are there other uses of var that does produce true for null?

Reviewer 2: @Thraka Nope, `thing is var newVariableName` always succeeds, and always returns true.

But, `thing is string message` returns `false` if `thing` is `null`.

> One reason to use the `var` pattern is to create a temporary variable within a Boolean expression. For example:

Reviewer: ```suggestion
You can use the `var` pattern to create a temporary variable within a Boolean expression, as the following example shows:
```

[!code-csharp[is#8](../../../../samples/snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs#8)]

> Here the temporary variable is used to store the result of an expensive operation. The variable can then be used multiple times.

Reviewer: ```suggestion
In the preceding example, the temporary variable is used to store the result of an expensive operation. The variable can then be used multiple times.
```
